packages/drivers/odsp-driver-definitions/src/resolvedUrl.ts

```diff
@@ -12,14 +12,53 @@ export interface IOdspUrlParts {
 }
 
 /**
+ * @deprecated Use SharingLinkKind type instead.
  * Type of shareLink requested/created when creating the file for the first time.
- * At the time of adding this comment (Sept/2021) ODSP only supports creation of CSL links
- * when provided as a request parameter with the /snapshot api call.
- * In future, we can add more types here.
 */
 export enum ShareLinkTypes {
     csl = "csl",
 }
+
+/**
+ * Sharing scope of the share links created for a file.
+*/
+export enum SharingLinkScope {
+    organization = "organization",
+    users = "users",
+    anonymous = "anonymous",
+    default = "default",
+}
+
+/**
+ * View/edit permission role for a sharing link.
+ */
+export enum SharingLinkRole {
+    view = "view",
+    edit = "edit",
+}
+
+/**
+ * Defines the permissions scope for a share link requested to be created during the creation the file in ODSP.
+ * Providing these properties to the /snapshot api will also create and return the requested kind of sharing link.
+*/
+export interface SharingLinkKind {
+    linkScope: SharingLinkScope;
+    /*
+     * If this parameter is not provided, the API will default to "edit" links (provided
+     * a valid createLinkScope setting is given).
+    */
+    linkRole?: SharingLinkRole;
+}
+
+/**
+ * Sharing link data received from the /snapshot api response.
+ */
+export interface SharingLink {
+    webUrl: string;
+    scope?: SharingLinkScope;
```

Comment:
This interface captures the response object that we will get from ODSP. Unfortunately, it looks different from ShareLinkKind type, since they provide 'scope' and 'type' (for role) values in the response. I could extract the response and transform it, but I am trying to keep a loose coupling, where I assign the shareLink response as is, to avoid any future driver code changes if the response changes even slightly.

Another option is to not define this interface at all and design the `sharingLink: any` kind. Let me know what you think.

Reply:
We can keep the interface but I would suggest adding word "Response" in the name to clearly suggest that it is response for a network call.

Reply:
Updated the `SharingLink` interface to inherit from `SharingLinkKind`.